src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
+    private File fileForRead;
+    private File fileForWrite;
+
+    void getStatistic(String fromFileName, String toFileName) {
+        fileForRead = new File(fromFileName);
+        fileForWrite = new File(toFileName);
+        String dataString = readFile(fileForRead);
+        String stringDataToFile = generateReport(dataString);
+        writeToFile(fileForWrite, stringDataToFile);
+    }
+
+    private String readFile(File file) {
+        StringBuilder stringBuilderForFilesData = new StringBuilder();
+        try {
+            FileReader fileReader = new FileReader(fileForRead);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            int valueResultMethod = bufferedReader.read();
```

Comment:
please use more obvious naming for variable, like below or your own option
```suggestion
            int currentSymbol = bufferedReader.read();
```